### DISCUSSION TITLE
Activate three events from one button

### DISCUSSION THREAD

**Discussion by Robjones6146** [artisan-roaster-scope/artisan#1239] [General]
I am trying to activate three events with one button click.

I read that: "You can put a sequence of commands separated by semicolons"

I tried that, but could not get the set command to work: (Linear actuator open, Relay on, Linear actuator open)
IO Command   |   vel(0,-1,684949:2) ; set(0,1,685111:3) ; vel(0,-1,685111:4) 

 I get this message: 100 11:46:14.185 Failed to set(0, 1, 685111:3)

Can anyone help with this, please? Where am I going wrong?

Thank you.

**Comment by roasterdave** [comment]
The set command does not look right.  It should have either a 0 or 1 in the second position.  Double check the channel number too.

From Config>> Events>> Buttons tab>> Help:
`set(c,b[,sn])      PHIDGET Binary Output: switches channel c off (b=0) and on (b=1)`

**Reply by Robjones6146** [reply]
It works fine by itself as a button. The second position is 1. -- set(0,1,685111:3)  -- set channel 0 to ON for serial# HUB channel 3

**Reply by roasterdave** [reply]
Oops, I must need new glasses.  I saw a period following the 1 where indeed there is a comma.

**Reply by Robjones6146** [reply]
No problem, I have the same issue wearing glasses. Any other ideas to trigger other buttons from one button?

**Comment by MAKOMO** [comment]
I tested your command (with different serial/port numbers) to match my setup and it works as expected. Do you have a typo in your serial/port number? Note that the port number 3 should indicate the VINT HUB port the module is connected too, while the channel number 0 in `set(0,1,685111:3)` addresses channel 0 on the module and the serial number `685111` is the serial number of the VINT Hub. The error message just indicates that Artisan couldn't find an attached channel with the specified serial number port combination to talk to.

There is the `Multiple Event` action which expects a comma separated list of button IDs in the Documentation field. It will fire all referred buttons.

**Reply by Robjones6146** [reply]
Thank you. My mistake. Filemaker was open now, I can't remember if that was the case before, but it works now. I use the Filemaker database to capture all the bean weights, print labels, weigh beans into bags, schedule roasts, and much, much more. You can only connect to one software at a time... I tried the Multiple Event which I had read about before, but can't remember where. It is a better solution for me as the Command could be VOUT or IO. The Value is also a bit confusing for me as there are obviously different values, although it still works.
Thanks for the help, much appreciated.

**Comment by Robjones6146** [comment]
Thanks, Marko.
